component/src/stake/component.rs

```diff
@@ -754,14 +765,49 @@ impl Staking {
             }
         }
 
-        // Update the consensus key lookup, in case the validator rotated their
-        // consensus key.
-        self.state
-            .put_domain(
-                state_key::validator_id_by_consensus_key(&validator.consensus_key).into(),
-                validator.identity_key.clone(),
-            )
-            .await;
+        // If the consensus key changed, then the old consensus key's voting power drops
+        // to 0, and the new one isn't recorded (since a newly added validator can
+        // never be active, and Tendermint can't handle receiving a new validator with 0
+        // power).
```

Comment:
I'm not sure this is true. What happens when someone rotates the consensus key of a currently active validator?

Reply:
The comment is inaccurate, what would happen in that case is that the new consensus key would get a voting power assigned during `end_epoch` (which would occur after the validator definitions are handled), so everything should work out correctly.